packages/block-library/src/image/badgeable.native.js

```diff
@@ -0,0 +1,24 @@
+/**
+ * External dependencies
+ */
+import React from 'react';
+import { Text, View } from 'react-native';
+
+/**
+ * Internal dependencies
+ */
+import styles from './styles.scss';
+
+const Badge = ( { text } ) => (
+	<View style={ styles.badge }>
+		<Text style={ styles.badgeText }>{ text }</Text>
+	</View>
+);
+const Badgeable = ( { text, children, show = true } ) => (
```

Comment:
I agree it's nice not to have to decide where to add the badge when rendering a component 👍
Could you explain the benefit of this approach in your PR description, vs say an HOC or simply adding the badge in each component?
Asking because we may also want to write a documentation (README.md) about this component.

Reply:
Sure!